### PR TITLE
[Click to Call]create helper for outbound callcomposite

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Creates helper function to load `outboundCallComposite` in
- Angular
- Vue
- Other frameworks that use ES6 imports
# Why

allows for more cross platform support with javascript

# How Tested

built bundle and tested locally with Vue test app
image
Built bundle and tested with Angular app
image

### CODE REVIEW

File: packages/acs-ui-javascript-loaders/src/outboundCallCompositeLoader.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import React from 'react';
+import { createRoot } from 'react-dom/client';
```
Comment (JamesBurnside) [inline packages/acs-ui-javascript-loaders/src/outboundCallCompositeLoader.ts:5]: Hmm this depends on the react version being used, i.e. will break if react 16 is used

File: common/config/rush/variants/stable/pnpm-lock.yaml [hunk 2]
```diff
@@ -542,7 +555,7 @@ packages:
     resolution: {integrity: sha512-QkBXwGgaoC2GtGZRoma6kv7Szfv06khvhFav67ZExau2RaXzy8MpHSMO2PNoP2XtmQphJQRHFfg77Bq731Yizw==}
     engines: {node: '>=6.9.0'}
     dependencies:
-      '@babel/types': 7.24.5
+      '@babel/types': 7.24.0
```
Comment (emlynmac) [inline common/config/rush/variants/stable/pnpm-lock.yaml:558]: Probably don't want a downgrade here?

File: common/config/rush/variants/stable/pnpm-lock.yaml [hunk 3]
```diff
@@ -604,9 +617,9 @@ packages:
     dependencies:
       '@babel/core': 7.24.7
       '@babel/helper-compilation-targets': 7.23.6
-      '@babel/helper-module-imports': 7.24.3
+      '@babel/helper-module-imports': 7.22.15
```
Comment (emlynmac) [inline common/config/rush/variants/stable/pnpm-lock.yaml:620]: why the downgrade here?

File: packages/acs-ui-javascript-loaders/.gitignore [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Build dirs
+build/
+node_modules/
+dist/
```
Comment (emlynmac) [inline packages/acs-ui-javascript-loaders/.gitignore:4]: does preprocessed need adding here too?

File: packages/acs-ui-javascript-loaders/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# @internal/acs-ui-javascript-loaders
```
Comment (emlynmac) [inline packages/acs-ui-javascript-loaders/README.md:1]: Should this package be a separate release so as to not make the main react bundle larger?

File: packages/acs-ui-javascript-loaders/package.json [hunk 6]
```diff
@@ -0,0 +1,71 @@
+{
+  "name": "@internal/acs-ui-javascript-loaders",
```
Comment (emlynmac) [inline packages/acs-ui-javascript-loaders/package.json:2]: Should this package be a separate release so as to not make the main react bundle larger?

File: packages/acs-ui-javascript-loaders/src/outboundCallCompositeLoader.ts [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+const reactVersion = React.version;
+const parseReactVersion = (version: string | undefined): number[] => {
+  if (!version) {
+    return [];
+  }
+  return version.split('.').map((v) => parseInt(v));
+};
+if (parseReactVersion(reactVersion)[0] && parseReactVersion(reactVersion)[0] < 18) {
+  throw new Error(
+    'React version is less than 18. Please upgrade to React 18 or alternatively checkout how to use our composites directly here: https://azure.github.io/communication-ui-library/?path=/docs/quickstarts-composites--page'
```
Comment (emlynmac) [inline packages/acs-ui-javascript-loaders/src/outboundCallCompositeLoader.ts:15]: Should we set the package peer deps to reflect this v18 requirement?

**Review by emlynmac — COMMENTED** [review]
Should this new package be a freestanding / released package?

File: common/config/rush/variants/stable/pnpm-lock.yaml (hunk 2 above)
Comment (dmceachernmsft) [reply common/config/rush/variants/stable/pnpm-lock.yaml:558]: you would be right about these downgrades. having to merge main into this guy has been tricky managing this stuff will update all the down grades

File: packages/acs-ui-javascript-loaders/.gitignore (hunk 4 above)
Comment (dmceachernmsft) [reply packages/acs-ui-javascript-loaders/.gitignore:4]: yes it should for sure!

**Pull request closed without merging**